### PR TITLE
W-15207052-Object-store-key-character-restrictions

### PR DESCRIPTION
**Pull request by hannanelson**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/_partials/osv2-faq.adoc [hunk 1]
```diff
@@ -13,7 +13,7 @@ On-premises Mule applications can access Object Store v2 only via the Object Sto
 //end::faqOnPrem[]
 
 //tag::faqKeyChars[]
-Object Store v2 doesn't support using pipe (`|`) characters in keys.
+Object Store v2 doesn't support using pipe (`|`) characters or spaces in keys.
```
Comment (hannanelson) [inline modules/ROOT/pages/_partials/osv2-faq.adoc]:
```suggestion
Object Store v2 doesn't support using spaces or pipe (`|`) characters in keys.
```

**Review by somal-vora — DISMISSED** [review]
LGTM

File: modules/ROOT/pages/_partials/osv2-faq.adoc [hunk 2]
```diff
@@ -13,7 +13,7 @@ On-premises Mule applications can access Object Store v2 only via the Object Sto
 //end::faqOnPrem[]
 
 //tag::faqKeyChars[]
-Object Store v2 doesn't support using pipe (`|`) characters in keys.
+Object Store v2 doesn't support using pipe (`|`) characters in keys and converts any spaces to plus (`+`) characters.
```
Comment (GouthamiKondakindi) [inline modules/ROOT/pages/_partials/osv2-faq.adoc]: Only OSv2 plugin does this. The REST API doesn't do this conversion, it takes user input key and stores it as is in the backend.

**Review by GouthamiKondakindi — APPROVED** [review]
LGTM

**Review by Crispy-Salesforce — APPROVED** [review]

**Pull request merged**